llvm/lib/Analysis/InstructionSimplify.cpp

```diff
@@ -6549,6 +6549,10 @@ Value *llvm::simplifyBinaryIntrinsic(Intrinsic::ID IID, Type *ReturnType,
     if (match(Op0, m_ImmConstant()))
       std::swap(Op0, Op1);
 
+    // Propagate poison.
+    if (isa<PoisonValue>(Op1))
```

Comment:
It has been handled by `simplifyCall->tryConstantFoldCall->ConstantFoldCall`:
https://github.com/llvm/llvm-project/blob/acc40ed0a747ad8f5dbb93788da7ce104c36c0a3/llvm/lib/Analysis/ConstantFolding.cpp#L3181-L3188

Reply:
Oh, sorry for the noise :(

Reply:
Ah thanks for showing this, that explains why two poison operands get folded.